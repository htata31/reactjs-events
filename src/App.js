import React, { Component } from 'react';
import './App.css';
import Header from './Header';
import Homepage from './Hompage';


class App extends Component{
  state = {
    contacts: []
  }
  componentDidMount() {
    fetch('http://jsonplaceholder.typicode.com/users')
    .then(res => res.json())
    .then((data) => {
      this.setState({ contacts: data })
    })
    .catch(console.log)
  }
  render(){
  return (
    <div>
      <Header title = 'React Bootstrap Meetup Application'/>
      <Homepage contacts={this.state.contacts} />
    </div>
  );
  }
}

export default App;
