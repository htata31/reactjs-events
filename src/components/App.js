import React, { Component } from 'react';
import './App.css';
import Header from './Header';
import Homepage from './Hompage';

/**
 * Using "https://jsonplaceholder.typicode.com/users" api I have implemented the application
 */
class App extends Component{
  state = {
    contacts: []
  }
  componentDidMount() {
    fetch('https://jsonplaceholder.typicode.com/users')
    .then(res => res.json())
    .then((data) => {
      this.setState({ contacts: data })
    })
    .catch(console.log)
  }
  render(){
  return (
    <div>
      <Header title = 'React Bootstrap Meetup Application'/>
      <Homepage contacts={this.state.contacts} />
    </div>
  );
  }
}

export default App;
