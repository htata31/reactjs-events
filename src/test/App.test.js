import enzyme, {shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import React from 'react';
import toJson from 'enzyme-to-json';
import App from '../components/App';

enzyme.configure({ adapter: new enzymeAdapterReact16(), });

function getAppShallowWrapper() {
  return shallow(
    <App />
  );
}
describe('The App component', () => {
  test('Taking a snapshot of the app component', () => {
    const appWrapper = getAppShallowWrapper();

    expect(toJson(appWrapper)).toMatchSnapshot();
  });

});
