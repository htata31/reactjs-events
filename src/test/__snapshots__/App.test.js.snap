// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The HomePage component Taking a snapshot of the app component 1`] = `
<div>
  <Header
    title="React Bootstrap Meetup Application"
  />
  <Homepage
    contacts={Array []}
  />
</div>
`;
